Album/ViewControllers/MapControllers/FriendMapViewController.m

```diff
@@ -89,6 +105,21 @@ - (void)fetchMarkers {
             // Store the posts, update count
             NSLog(@"Successfully fetched markers!");
             self.markerArr = (NSMutableArray *)pins;
+            for (PFObject *pin in pins) {
+                if (!self.placeToPins[pin[@"placeName"]]) {
+                    [self.placeToPins setObject:[[NSMutableArray alloc]init] forKey:pin[@"placeName"]];
+                }
+                [self.placeToPins[pin[@"placeName"]] addObject:pin];
+                // Save images of the specific pin to the cache data structure
+                [self imagesFromPin:pin.objectId withBlock:^(NSArray *_Nullable images, NSError *_Nullable error) {
+                    if (images != nil) {
+                        // Set image of the info window to first in the array
+                        [self.pinImages setObject:images forKey:pin.objectId];
+                    } else {
+                        NSLog(@"%@", error.localizedDescription);
```

Comment:
It would be better if we could handle the errors and let the user know in a better way.
maybe an alert

Reply:
Done!

Reply:
it does not seems done here, I think you did it in the signup screen.